### PR TITLE
initial implementation

### PR DESCRIPTION
**Pull request by panentheos**
Provides an initial implementation of the audio service. Starts a web server with one endpoint that performs TTS and caching. Secures access with a pre-shared API key. Allows stubbing out the actual Polly calls in dev/staging to save money.

Things needing future improvement:
* API docs
* Explicit input validation and helpful error messages

### CODE REVIEW

File: lib/web_api.ex [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+  plug(Plug.Parsers, parsers: [{:json, json_decoder: Jason}])
+  plug(:match)
+  plug(:enforce_api_key)
+  plug(:dispatch)
+
+  post "/tts" do
+    %{"text" => text, "lang" => lang} = conn.body_params
+
+    voice_id =
```
Comment (cmaddox5) [inline lib/web_api.ex:13]: Could this be something that can come from the request and default to these if absent? Might be thinking a bit too far in the future here, but If the Screens app were to eventually use this service, they use a different voice. Could give a bit more flexibility to apps using the service.

File: lib/web_api.ex [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+            send_resp(conn, 400, body)
+        end
+    end
+  end
+
+  defp enforce_api_key(conn, _opts) do
+    api_key =
+      Enum.find_value(conn.req_headers, fn {key, value} -> if(key == "x-api-key", do: value) end)
+
+    if api_key != Application.get_env(:watts, :api_key) do
```
Comment (cmaddox5) [inline lib/web_api.ex:79]: What would be the process required for requesting the production key if an app deploys code using this? Should devs ask you or request the secret from infra?

**Review by cmaddox5 — APPROVED** [review]
Looks great! Only thing I can think of that's missing is deployment related stuff, but definitely doesn't block an initial commit. Had a couple of questions, but they are non-blocking/optional.

File: lib/web_api.ex (hunk 1 above)
Comment (panentheos) [reply lib/web_api.ex:13]: Makes sense. I wrote it like this thinking we wanted to standardize on a particular voice, but if we want different ones for different apps then we'll need to offer that flexibility.

File: lib/web_api.ex (hunk 2 above)
Comment (panentheos) [reply lib/web_api.ex:79]: I'm assuming that the key will live in secrets manager, so we can inject it as an environment variable to any app that needs to make calls.

**Pull request merged**